src/cmd/sanity_test/sanity_tests.go

```diff
@@ -112,6 +119,76 @@ func checkEmailRestoration(client *msgraphsdk.GraphServiceClient, testUser, fold
 				*restore.GetTotalItemCount())
 			os.Exit(1)
 		}
+
+		checkAllSubFolder(user, restore, "", messageCount)
+	}
+}
+
+func getAllSubFolder(
+	user *users.UserItemRequestBuilder,
+	r models.MailFolderable,
+	parentFolder string,
+	messageCount map[string]int32,
+) {
+	folder := user.MailFoldersById(*r.GetId())
```

Comment:
Prefer ptr.Val for graph object gets, and storing them at the top for re-use.
```suggestion
        var (
            folderID = ptr.Val(r.GetId())
        )

	folder := user.MailFoldersById(folderID)
```